commands/operator-sdk/cmd/scorecard/scorecard.go

```diff
@@ -0,0 +1,195 @@
+	olmApi.AddToScheme(scheme)
+	dynamicDecoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
+	// if a user creates a new CRD, we need to be able to reset the rest mapper
+	// temporary kubeclient to get a cached discovery
+	kubeclient, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to get a kubeclient: %v", err)
+	}
+	cachedDiscoveryClient := cached.NewMemCacheClient(kubeclient.Discovery())
+	restMapper = restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
+	restMapper.Reset()
+	runtimeClient, _ = client.New(kubeconfig, client.Options{Scheme: scheme, Mapper: restMapper})
+	err = createFromYAMLFile(SCConf.GlobalMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create global resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.NamespacedMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create namespaced resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.CrMan, true)
+	if err != nil {
+		return fmt.Errorf("failed to create cr resource: %v", err)
+	}
+	obj := unstructured.Unstructured{}
+	obj.SetAPIVersion(apiversion)
+	obj.SetKind(kind)
+	if SCConf.BasicTests {
+		fmt.Printf("Checking spec and stat\n")
+		err = checkSpecAndStat(runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Checking status update\n")
+		err = checkStatusUpdate(runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		logs, err := writingIntoCRsHasEffect(obj)
+		if err != nil {
+			return err
+		}
+
+		/* this print out a lot of ugly logs and should only be used for debugging
+		fmt.Printf("Scorecard Proxy Logs: %v\n", logs)
+		*/
+		// this is just to prevent a "declared but not used" error
+		strings.Contains(logs, "test")
+	}
+	if SCConf.OlmTests {
+		yamlSpec, err := ioutil.ReadFile(SCConf.CSVPath)
+		if err != nil {
+			return fmt.Errorf("failed to read csv: %v", err)
+		}
+		rawCSV, _, err := dynamicDecoder.Decode(yamlSpec, nil, nil)
+		if err != nil {
+			return fmt.Errorf("%v", err)
+		}
+		csv := &olmApi.ClusterServiceVersion{}
+		switch o := rawCSV.(type) {
+		case *olmApi.ClusterServiceVersion:
+			csv = o
+		default:
+			return fmt.Errorf("provided yaml file not of ClusterServiceVersion type")
+		}
+		fmt.Printf("Checking for CRD resources\n")
+		crdsHaveResources(csv)
+		fmt.Printf("Checking for existence CR example\n")
+		annotationsContainExamples(csv)
+		fmt.Printf("Checking spec descriptors\n")
+		err = specDescriptors(csv, runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Checking status descriptors\n")
+		err = statusDescriptors(csv, runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+	}
+	var totalEarned, totalMax int
+	for _, testType := range []string{basicOperator, olmIntegration, goodTenant} {
+		fmt.Printf("%s:\n", testType)
+		for _, test := range scTests {
+			if test.testType == testType {
+				fmt.Printf("\t%s: %d/%d\n", test.name, test.earnedPoints, test.maximumPoints)
```

Comment:
Nit: Can we add points to the end of this, just to make sense of the numbers?
```suggestion
				fmt.Printf("\t%s: %d/%d points\n", test.name, test.earnedPoints, test.maximumPoints)
```